pkg/deploy/registry/deploylog/rest.go

```diff
@@ -180,6 +186,31 @@ func (r *REST) Get(ctx kapi.Context, name string, opts runtime.Object) (runtime.
 	}, nil
 }
 
+// waitForExistingDeployment will use the timeout to wait for a deployment to appear.
+func (r *REST) waitForExistingDeployment(namespace, name string) (*kapi.ReplicationController, error) {
+	var (
+		target *kapi.ReplicationController
+		err    error
+	)
+
+	condition := func() (bool, error) {
```

Comment:
Is there a scenario where we know up front this will never happen (paused deployment, no triggers) that we should not wait?  Or is the user expected to handle that?  Since logs can be called from any endpoint, is the following valid?

```
oc logs dc/foo &
oc deploy --latest
```

?